### PR TITLE
http: add custom header support for sha1 fingerprint of peer cert

### PR DESCRIPTION
**Pull request by ramaraochavali**
Commit Message:add custom header support for sha1 fingerprint of peer cert
Additional Description: Adds support for `DOWNSTREAM_PEER_FINGERPRINT_1`  that sends SHA1 finger print of peer cert as a custom header.
Risk Level: Low, only when enabled via config
Testing: Added tests
Docs Changes: Updated
Release Notes: Updated

### CODE REVIEW

File: source/extensions/transport_sockets/tls/ssl_socket.cc [hunk 1]
```diff
@@ -371,6 +371,24 @@ const std::string& SslSocketInfo::sha256PeerCertificateDigest() const {
...
+  if (!cached_sha_1_peer_certificate_digest_.empty()) {
+    return cached_sha_1_peer_certificate_digest_;
+  }
+  bssl::UniquePtr<X509> cert(SSL_get_peer_certificate(ssl()));
+  if (!cert) {
+    ASSERT(cached_sha_1_peer_certificate_digest_.empty());
+    return cached_sha_1_peer_certificate_digest_;
+  }
+
+  std::vector<uint8_t> computed_hash(SHA_DIGEST_LENGTH);
```
Comment (ramaraochavali) [inline source/extensions/transport_sockets/tls/ssl_socket.cc:384]: @lizan Not fully familiar with open ssl code - Is using `SHA_DIGEST_LENGTH` for `SHA1` digest correct?

**Comment by PiotrSikora** [comment]
...but why?

**Comment by ramaraochavali** [comment]
The reason is our core application uses `sha1` (in the process of  migrating to `sha256`) and there are still customers using `sha1`  and database still has `sha1` . It takes lot of time to migrate every one to `sha256` and for BC till the migration happens we need `sha1` support.

File: source/extensions/transport_sockets/tls/ssl_socket.cc (hunk 1 above)
Comment (lizan) [reply source/extensions/transport_sockets/tls/ssl_socket.cc:384]: Yes I think that's right.

**Comment by ramaraochavali** [comment]
@lizan Can you please merge this or are  you waiting for any other review?

File: docs/root/version_history/current.rst [hunk 2]
```diff
@@ -38,6 +38,7 @@ New Features
 
 * ext_authz filter: added support for emitting dynamic metadata for both :ref:`HTTP <config_http_filters_ext_authz_dynamic_metadata>` and :ref:`network <config_network_filters_ext_authz_dynamic_metadata>` filters.
 * grpc-json: support specifying `response_body` field in for `google.api.HttpBody` message.
+* http: added support for :ref:`%DOWNSTREAM_PEER_FINGERPRINT_1% <config_http_conn_man_headers_custom_request_headers>` as custom header.
```
Comment (mattklein123) [inline docs/root/version_history/current.rst:41]: You also added this for access log also. Can you release note that?

**Comment by ramaraochavali** [comment]
@mattklein123 updated the release notes. Can you PTAL when you get chance?

**Pull request merged**